inference-engine/src/mkldnn_plugin/nodes/ctc_greedy_decoder_seq_len.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "base.hpp"
+#include "ie_parallel.hpp"
+
+#include <vector>
+#include <string>
+
+namespace InferenceEngine {
+namespace Extensions {
+namespace Cpu {
+
+class CTCGreedyDecoderSeqLenImpl: public ExtLayerBase {
+public:
+    explicit CTCGreedyDecoderSeqLenImpl(const CNNLayer* layer) : mergeRepeated_(true) {
+        std::string errPrefix = "CTCGreedyDecoderSeqLen layer with name '" + layer->name + "' ";
+        if (layer->insData.size() < 2 || layer->insData.size() > 3)
+            THROW_IE_EXCEPTION << errPrefix << "has invalid number of input edges: " << layer->insData.size();
+        if (layer->outData.size() != 2)
+            THROW_IE_EXCEPTION << errPrefix << "has invalid number of outputs edges: " << layer->outData.size();
+
+        auto inData = layer->insData[DATA_INDEX].lock();
+        auto sequenceLenData = layer->insData[SEQUENCE_LENGTH_INDEX].lock();
+        if (!inData || !sequenceLenData)
+            THROW_IE_EXCEPTION << errPrefix << "has nullable inputs.";
+        if (inData->getTensorDesc().getDims()[0] != sequenceLenData->getTensorDesc().getDims()[0])
+            THROW_IE_EXCEPTION << errPrefix << "has invalid input shapes.";
+        if (inData->getTensorDesc().getPrecision() != Precision::FP32 &&
+                inData->getTensorDesc().getPrecision() != Precision::BF16)
+            THROW_IE_EXCEPTION << errPrefix << "has unsupported 'data' input precision: " << inData->getTensorDesc().getPrecision();
+        if (sequenceLenData->getTensorDesc().getPrecision() != Precision::I32)
```

Comment:
What about I64? Anyway you are creating config for I32 so the plug-in will handle data type conversion automatically.

Reply:
I64 is converted to I32 during transformations.

Reply:
Fair enough, but I just want to avoid refactoring of each node (like we did for BF16 support) once we will switch off this transformation.

Reply:
Fixed